cupy/core/core.pyx

```diff
@@ -3196,6 +3206,8 @@ cpdef ndarray _take(ndarray a, indices, li=None, ri=None, ndarray out=None):
             raise TypeError('Output dtype mismatch')
         if out.shape != out_shape:
             raise ValueError('Output shape mismatch')
+    if a.size == 0 and out.size != 0:
```

Comment:
The condition is different from the error message.  It seems the following case is not covered.
```
>>> cupy.empty((0, 0)).take([1], axis=0)
array([], shape=(1, 0), dtype=float64)
```

Reply:
Thanks. but, `cupy.take` behaves wrap mode for performance.

Reply:
In this example, wrap mode means `cupy.take` computes `a[1%0]`, doesn't it?

Reply:
I wrote the issue numpy/numpy#11677.